bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -40,8 +40,12 @@ class MsgType {
 };
 
 struct BCStateTranBaseMsg {
-  BCStateTranBaseMsg(uint16_t type) : type(type) {}
+  BCStateTranBaseMsg(uint16_t type) : BCStateTranBaseMsg(type, false) {}
+  BCStateTranBaseMsg(uint16_t type, bool isIncomingMsg) : type(type), is_incoming_msg_(isIncomingMsg) {}
   uint16_t type;
+  // this flag is for monitoring messages buffer allocs and releases so it's mutable eventhough incoming ST msgs objs
+  // are const during handling
+  mutable bool is_incoming_msg_ = false;
```

Comment:
The size of boolean variables differs among OSs. In case this value passes through the network, it should be defined as uint8_t.